File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/authentication/DeleteAccountUseCase.java
package com.lattis.ellipse.domain.interactor.authentication;

import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.repository.UserRepository;

import javax.inject.Inject;

import io.reactivex.Observable;

public class DeleteAccountUseCase extends UseCase<Boolean> {

    private UserRepository userRepository;

    @Inject
    DeleteAccountUseCase(ThreadExecutor threadExecutor,
                         PostExecutionThread postExecutionThread,
                         UserRepository userRepository) {
        super(threadExecutor, postExecutionThread);
        this.userRepository = userRepository;
    }

    @Override
    protected Observable<Boolean> buildUseCaseObservable() {
        return this.userRepository.deleteUserAccount();
    }

}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/user/DeleteAccountUseCase.kt
package com.lattis.domain.usecase.user

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.UserRepository
import com.lattis.domain.usecase.base.UseCase
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class DeleteAccountUseCase @Inject
constructor(threadExecutor: ThreadExecutor,
            postExecutionThread: PostExecutionThread,
            private val userRepository: UserRepository
) : UseCase<Boolean>(threadExecutor, postExecutionThread) {

    override fun buildUseCaseObservable(): Observable<Boolean> {
        return userRepository.deleteAccount()
    }
}